src/with-component.js

```diff
@@ -2,9 +2,10 @@ import { render } from 'preact';
 import { HTMLElement } from './util';
 import { withProps } from './with-props';
 import { withRender } from './with-render';
+import { withUnique } from './with-unique';
 
 export function withComponent (Base = HTMLElement) {
-  return class extends withRender(withProps(Base)) {
+  return class extends withUnique(withRender(withProps(Base))) {
```

Comment:
so now user created components will get 4 levels of prototype chain by default. hmm :D let's see how this goes from performance perspective

```
UserClassComponent
  --> uniqueClass
    --> renderClass
      --> propsClass
        --> HTMLElement
```

Let's see how this goes with perf as we discussed already. 
Just thinking what could we use if we ran into perf problems? I guess just directly mutating user `UserClassComponent` class?